### PR TITLE
MINOR: doc changes for KIP-138

### PR DESCRIPTION
**Pull request by guozhangwang**
1. Core concepts (added the stream time definition), upgrade guide and developer guide.
2. Related Java docs changes.

### CODE REVIEW

**Comment by guozhangwang** [comment]
@mihbor @dguy @mjsax @bbejeck @miguno for reviews.

Also while working on the doc changes, I feel that `SYSTEM_TIME` string name may be less clear compare to `WALL_CLOCK_TIME`. so I'd like to have a very minor addendum on KIP-138 to rename this string name accordingly. WDYT?

**Comment by guozhangwang** [comment]
Jenkins failures are not related.

File: docs/streams/core-concepts.html [hunk 1]
```diff
@@ -95,14 +95,14 @@
...
-        Kafka Streams assigns a <b>timestamp</b> to every data record
-        via the <code>TimestampExtractor</code> interface.
-        Concrete implementations of this interface may retrieve or compute timestamps based on the actual contents of data records such as an embedded timestamp field
-        to provide event-time semantics, or use any other approach such as returning the current wall-clock time at the time of processing,
-        thereby yielding processing-time semantics to stream processing applications.
-        Developers can thus enforce different notions of time depending on their business needs. For example,
-        per-record timestamps describe the progress of a stream with regards to time (although records may be out-of-order within the stream) and
-        are leveraged by time-dependent operations such as joins.
+        Kafka Streams assigns a <b>timestamp</b> to every data record via the <code>TimestampExtractor</code> interface.
+        These per-record timestamps describe the progress of a stream with regards to time and are leveraged by	and are leveraged by time-dependent operations such as window operations.
```
Comment (bbejeck) [inline docs/streams/core-concepts.html]: extra "and are leveraged by"

File: docs/streams/core-concepts.html [hunk 2]
```diff
@@ -95,14 +95,14 @@
...
+        As a result, this time will only advance when there is new record arrived at the processor.
```
Comment (bbejeck) [inline docs/streams/core-concepts.html]: nit: "when a new record arrives at the processor"

**Comment by bbejeck** [comment]
@guozhangwang I agree with you.  Reading the section in the `upgrade-guide` where `wall-clock-time` is discussed then `SYSTEM_TIME` is referenced is a little unclear.  Not a big deal, but I think the changing to `WALL_CLOCK_TIME` is for the better.

**Comment by guozhangwang** [comment]
retest this please

**Review by dguy — COMMENTED** [review]
If you fix @bbejeck's suggestions, then i think it will be ready to go

**Comment by guozhangwang** [comment]
Cool, if people have no objections on the renaming, I will do this voting as KIP-138 addendum and rename these again in another PR.

cc @mihbor @dguy @mjsax @miguno

File: docs/streams/core-concepts.html (hunk 1 above)
Comment (guozhangwang) [reply docs/streams/core-concepts.html]: ack

File: docs/streams/core-concepts.html (hunk 2 above)
Comment (guozhangwang) [reply docs/streams/core-concepts.html]: ack

**Comment by guozhangwang** [comment]
> If you fix @bbejeck's suggestions, then i think it will be ready to go

Could you review this again? @dguy

**Comment by dguy** [comment]
LGTM

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**